packages/contracts/test/integration/virtualAppAgreement.spec.ts

```diff
@@ -0,0 +1,86 @@
+import * as ethers from "ethers";
+
+import * as Utils from "../../utils/misc";
+
+const web3 = (global as any).web3;
+const { unlockedAccount } = Utils.setupTestEnv(web3);
+
+import { AbstractContract, buildArtifacts, expect } from "../../utils";
+
+/*
+This test deploys VirtualAppAgreement.sol and FixtedResolutionApp.sol as a
+target to test the virtual app functionality.
+*/
+
+contract("Virtual App", (accounts: string[]) => {
+  let virtualAppAgreement: ethers.Contract;
+  let registry: ethers.Contract;
+  let delegateProxy: ethers.Contract;
+
+  // @ts-ignore
+  before(async () => {
+    virtualAppAgreement = await (await buildArtifacts.VirtualAppAgreement).deploy(
+      unlockedAccount
+    );
+    registry = await (await buildArtifacts.Registry).deploy(unlockedAccount);
+    delegateProxy = await (await AbstractContract.fromArtifactName(
+      "DelegateProxy"
+    )).deploy(unlockedAccount);
+  });
+
+  describe("test", () => {
+    it("test", async () => {
+      const fixedResolutionApp = await (await AbstractContract.fromArtifactName(
+        "FixedResolutionApp",
+        {
+          Transfer: buildArtifacts.Transfer
+        }
+      )).deployViaRegistry(unlockedAccount, registry);
+
+      await unlockedAccount.sendTransaction({
+        to: delegateProxy.address,
+        value: ethers.utils.parseEther("100")
+      });
+
+      const beneficiaries = [
+        ethers.Wallet.createRandom().address,
+        ethers.Wallet.createRandom().address
+      ];
+
+      const tx = virtualAppAgreement.interface.functions.delegateTarget.encode([
+        {
+          beneficiaries,
+          registryAddr: registry.address,
+          terms: {
+            assetType: 0,
+            limit: 0,
+            token: ethers.constants.AddressZero
+          },
+          expiry: 0,
+          target: fixedResolutionApp.cfAddress,
+          capitalProvided: ethers.utils.parseEther("10")
+        }
+      ]);
+
+      expect(
+        (await unlockedAccount.provider.getBalance(beneficiaries[0])).toString()
+      ).to.eq("0");
+      expect(
+        (await unlockedAccount.provider.getBalance(beneficiaries[1])).toString()
+      ).to.eq("0");
+
+      await delegateProxy.functions.delegate(
+        virtualAppAgreement.address,
+        tx,
+        Utils.HIGH_GAS_LIMIT
+      );
+
+      expect(
+        (await unlockedAccount.provider.getBalance(beneficiaries[0])).toString()
+      ).to.eq(ethers.utils.parseEther("5").toString());
+      expect(
+        (await unlockedAccount.provider.getBalance(beneficiaries[1])).toString()
+      ).to.eq(ethers.utils.parseEther("5").toString());
```

Comment:
This test case handles fixed values of 10 capitalProvided to a specific target app -- can you abstract it to run for other cases? Examples:

- target app being 0x0
- capital provided being 0 and resolution being nonzero
- capital provided being 0.99999 and resolution being 1 (or generally speaking, just not being `==`)
- expiry time not being reached yet
- terms limit not being met precisely
- other ones you can think of

Reply:
`limit` is not checked, btw - the `agreement.capitalProvided > resolution.value[0]` line does the same check